Aardvark/ARKEmailBugReporter.h

```diff
@@ -39,6 +39,37 @@
 @end
 
 
+@interface ARKEmailAttachment : NSObject
+
+- (nonnull instancetype)initWithFileName:(nonnull NSString *)fileName data:(nonnull NSData *)data dataMIMEType:(nonnull NSString *)dataMIMEType;
+
+- (nonnull instancetype)init NS_UNAVAILABLE;
++ (nonnull instancetype)new NS_UNAVAILABLE;
+
+@property (nonnull, nonatomic) NSString *fileName;
+
+@property (nonnull, nonatomic) NSData *data;
+
+@property (nonnull, nonatomic) NSString *dataMIMEType;
+
+@end
+
+
+@protocol ARKEmailBugReporterEmailAttachmentAdditionsDelegate <NSObject>
+
+@optional
```

Comment:
I feel like it would be fairly common to only want to implement one of these protocol methods (either adding an additional attachment, or filtering out log stores).  Do you have an alternative you would prefer?

Reply:
Default / standard implementations of each seem quite simple, so I'd just require adoptees to write both methods.

I'm following Swift's lead re optional delegate methods here; they feel like a deprecated language tool now. That, and optional methods mean we lose compile-time safety, which in a world where Swift's translation of Objc methods changes from release-to-release can cause runtime failures that could have been caught at compile-time.

Reply:
Fair enough.  Making both methods required.